taskfiles/k8s.yml

```diff
@@ -139,6 +139,8 @@ tasks:
         if ! grep -q "0" kuttl-exit-code; then
           echo "Kuttl returned exit code $(cat kuttl-exit-code). Creating artifacts tarball"
           tar -czf {{.KUTTL_ARTIFACTS_DIR | base}}.tar.gz {{.KUTTL_ARTIFACTS_DIR}}
+          buildkite-agent artifact upload {{.KUTTL_ARTIFACTS_DIR | base}}.tar.gz
```

Comment:
Nice work around! Just to make sure I understand, you're fixing the issue by instead manually running the upload from _within_ the container? Please note that either as a comment in this block or in the commit message. You've explained the issue but not the solution.

Side note, I wonder if we should try using task's `defer` clause and always upload artifacts. We could get rid of the `KUTTL_EXIT_CODE` hack here and then make a task for CI and task for local development.

Reply:
I will adjust comment in the k8s.yml file with the solution too.

I agree that it would be ideal to get rid of the hack never the less local development should not run buildkite-agent upload command.